### PR TITLE
GEOPY-487

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-487 - Implement ZTEM receiver/base station **

### CODE REVIEW

File: geoh5py/objects/surveys/electromagnetics/tipper.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+                except IndexError:
+                    warnings.warn(
+                        "Associated `base_stations` entity not set.", UserWarning
+                    )
+                    return None
+
+        return self._base_stations
+
+    @base_stations.setter
+    def base_stations(self, base: TipperBaseStations):
```
Comment (benk-mira) [inline geoh5py/objects/surveys/electromagnetics/tipper.py:121]: Add a validation (raise/error) if n_vertices of base stations not one or equal to n_vertices of receivers

**Review by benk-mira — CHANGES_REQUESTED** [review]
Looks good man!  Just a couple of cleanups and the handling of number of base stations we discussed

File: geoh5py/objects/surveys/electromagnetics/tipper.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        **kwargs,
+    ):
+        self._base_stations = base_stations
+
+        super().__init__(
+            object_type,
+            **kwargs,
+        )
+
+    # def copy(
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoh5py/objects/surveys/electromagnetics/tipper.py:55]: Done

File: geoh5py/objects/surveys/electromagnetics/tipper.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    def default_type_uid(cls) -> uuid.UUID:
+        """
+        :return: Default unique identifier
+        """
+        return cls.__TYPE_UID
+
+    @property
+    def default_receiver_type(self):
+        """
+        :return: Transmitter class
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoh5py/objects/surveys/electromagnetics/tipper.py:213]: Yip

File: geoh5py/objects/surveys/electromagnetics/tipper.py (hunk 1 above)
Comment (domfournier) [reply geoh5py/objects/surveys/electromagnetics/tipper.py:79]: Done

**Review by benk-mira — APPROVED** [review]
Great!

**Review by benk-mira — APPROVED** [review]
All good

**Pull request merged**